File: weathersdk/src/main/java/fr/ekito/myweatherlibrary/ws/WeatherWS.java
package fr.ekito.myweatherlibrary.ws;

import fr.ekito.myweatherlibrary.json.geocode.Geocode;
import fr.ekito.myweatherlibrary.json.weather.Weather;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by arnaud on 04/08/2016.
 */
public interface WeatherWS {

    @GET("/geocode")
    @Headers("Content-type: application/json")
    Observable<Geocode> geocode(@Query("address") String address);

    @GET("/weather")
    @Headers("Content-type: application/json")
    Observable<Weather> weather(@Query("lat") Double lat, @Query("lon") Double lon, @Query("lang") String lang);

}

File: weathersdk/src/main/java/fr/ekito/myweatherlibrary/ws/WeatherWS.kt
package fr.ekito.myweatherlibrary.ws

import fr.ekito.myweatherlibrary.json.geocode.Geocode
import fr.ekito.myweatherlibrary.json.weather.Weather
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query
import rx.Observable

/**
 * Created by arnaud on 04/08/2016.
 */
interface WeatherWS {

    @GET("/geocode")
    @Headers("Content-type: application/json")
    fun geocode(@Query("address") address: String): Observable<Geocode>

    @GET("/weather")
    @Headers("Content-type: application/json")
    fun weather(@Query("lat") lat: Double, @Query("lon") lon: Double, @Query("lang") lang: String): Observable<Weather>

}
